packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1065,6 +1087,24 @@ class TextInput {
     }
 
     final List<dynamic> args = methodCall.arguments as List<dynamic>;
+
+    if (method == 'TextInputClient.updateEditingStateWithTag') {
+      final TextInputClient client = _currentConnection._client;
+      assert(client != null);
+      if (client is AutofillTrigger) {
```

Comment:
Should all `TextInputClient` just have a currentAutofillScope getter and return `null` by default if they don't participate in AutoFill? (Not sure what the AutofillTrigger abstraction is adding)

Reply:
Functionality-wise I think it's the same. Adding `AutofillTrigger` makes it non-breaking?

Reply:
I just landed a very similar breaking change a few days ago. I think that makes breaking this less bad?

Reply:
I think I would prefer the cleaner API here without the type check. Implementing your own TextInputClient should be fairly un-common, so migration should be fairly easy?